src/components/plots/plot.ts

```diff
@@ -83,39 +83,23 @@ module Plottable {
 
     public remove() {
       super.remove();
-      this._datasetKeysInOrder.forEach((k) => this.removeDataset(k));
       this._scales().forEach((scale) => scale.offUpdate(this._renderCallback));
+      this.datasets().forEach((dataset) => this.removeDataset(dataset));
     }
 
     /**
-     * Adds a dataset to this plot. Identify this dataset with a key.
-     *
-     * A key is automatically generated if not supplied.
-     *
-     * @param {string} [key] The key of the dataset.
-     * @param {Dataset | any[]} dataset dataset to add.
+     * @param {Dataset} dataset
      * @returns {Plot} The calling Plot.
      */
-    public addDataset(dataset: Dataset | any[]): Plot;
-    public addDataset(key: string, dataset: Dataset | any[]): Plot;
-    public addDataset(keyOrDataset: any, dataset?: any): Plot {
-      if (typeof(keyOrDataset) !== "string" && dataset !== undefined) {
-        throw new Error("invalid input to addDataset");
-      }
-      if (typeof(keyOrDataset) === "string" && keyOrDataset[0] === "_") {
-        Utils.Methods.warn("Warning: Using _named series keys may produce collisions with unlabeled data sources");
-      }
-      var key  = typeof(keyOrDataset) === "string" ? keyOrDataset : "_" + this._nextSeriesIndex++;
-      var data = typeof(keyOrDataset) !== "string" ? keyOrDataset : dataset;
-      dataset = (data instanceof Dataset) ? data : new Dataset(data);
-
+    public addDataset(dataset: Dataset) {
```

Comment:
It's called `addDataset()`, so I think it's safe to assume that it adds a `Dataset`.